pallets/elections/src/lib.rs

```diff
@@ -279,6 +402,28 @@ pub mod pallet {
 
             Ok(())
         }
+
+        fn kick_out_underperformed_non_reserved_validators() {
+            let to_be_kicked_validators =
+                ToBeKickedOutFromCommittee::<T>::iter_keys().collect::<BTreeSet<_>>();
+            if !to_be_kicked_validators.is_empty() {
+                let non_reserved_validators = NextEraNonReservedValidators::<T>::get()
+                    .into_iter()
+                    .collect::<BTreeSet<_>>();
+                let non_reserved_validators_len = non_reserved_validators.len();
+                let non_reserved_in_order = non_reserved_validators
+                    .into_iter()
+                    .filter(|v| !to_be_kicked_validators.contains(v))
+                    .collect::<Vec<_>>();
+                if non_reserved_in_order.len() < non_reserved_validators_len {
```

Comment:
In this way, as it is coded I think it's correct because when we add someone to `ToBeKickedOutFromCommittee` (either manually in `kick_out_from_committee` tx or via automatic logic) we check whether node is in _next_ era non_reserved node set. The version you suggest was actually my first choice indeed, but then I've come to the conclusion there's `reserved` nodes that might eventually get into `ToBeKickedOutFromCommittee`, yet they won't be kicked actually and they will "magically" disappear from that list with no reason, which might be confusing to the user on UI side (we're going to display contents of `ToBeKickedOutFromCommittee`. 

I'll try to include information about the reason in the event though which is a good suggestion.

Reply:
But you're right it's not clear with the code for the reader - I don't need to check therefore intersection of `to_be_kicked_validators` and next era non_reserved in kick_out_underperformed_non_reserved_validators, let me simplify that.

Reply:
Ah no, I need to do that check as well. This is because there's a corner case scenario (not that unlikely though) in which someone modifies next era non_reserved via `change_validators`, so that for example remove validator `V` from that list which is also in `ToBeKickedOutFromCommittee`. Without a current check, we'll emit a false positive event that we're kicking `V` from the committee.

Reply:
I've included reason in the event

Reply:
Discussed offline, I'll change this to not check non reserved nodes when added to kick out list